OCI connections microcopy updates

Griffin-Sullivan opened the pull request:
<!--- If this is a non-code change, this template is not required; reference any issues or top-level descriptions as needed -->
<!--- All code change PRs should relate to an issue, reference it here; see example below -->
<!--- https://issues.redhat.com/browse/RHOAIENG-123456 -->
https://issues.redhat.com/browse/RHOAIENG-20231
## Description
Microcopy updates for OCI ModelCars on Connections and serving modals. https://github.com/opendatahub-io/odh-dashboard/pull/3868 covers part of the microcopy that is missing here.
<!--- Describe your changes in detail; the what, the why, any findings, etc -->
<!--- Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->

## How Has This Been Tested?
Checking screens against figma and microcopy doc
![Screenshot From 2025-03-13 10-41-09](https://github.com/user-attachments/assets/f2ca3f00-1ab8-4f90-bdbd-00bc09ee9979)
![Screenshot From 2025-03-13 10-41-20](https://github.com/user-attachments/assets/04dab6c8-0ef8-4fab-9021-88f155dbfc3e)
![Screenshot From 2025-03-13 10-41-42](https://github.com/user-attachments/assets/d03a5af1-932a-4d02-bd68-a485f3a5b493)
![Screenshot From 2025-03-13 10-41-58](https://github.com/user-attachments/assets/5c2883df-4b1f-4f2a-a5d6-3d5597a72b76)
![Screenshot From 2025-03-13 10-42-22](https://github.com/user-attachments/assets/e8a58ec9-1161-48dc-86ac-960b15d9d686)


<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Test Impact
Update a test with the right modal title
<!--- What tests have you done to cover the implemented functionality -->
<!--- If tests are not applicable, explain why here -->

## Request review criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 
<!--- You can ignore these if you are doing manifest, backend, internal logic, etc changes; aka non-UI / visual changes -->
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

Griffin-Sullivan commented on frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ConnectionOciPathField.tsx:
```diff
@@ -42,36 +43,45 @@ const ConnectionOciPathField: React.FC<ConnectionOciPathFieldProps> = ({
-            </HelperTextItem>
-            <HelperTextItem>
-              Example: registry.connect.redhat.com/redhat-test/test-mymodel:v3.2
+              Provide the location of the model. This is not part of your connection instance.
```
The spacing on this help text looks off since there's such a far gap from the title. Maybe I shouldn't use the FormSection title and just add a normal title with Helper text?

emilys314 replied:
Yeah it does look like a large gap from the title. Does this have to be a <HelperText/>? Or can that be removed and this become regular html text?

Griffin-Sullivan replied:
I guess it doesn't have to be I was just using it to match the rest of the form. Let me try some other ways around it

Griffin-Sullivan commented:
@simrandhaliw OCI microcopy is ready for review!

emilys314 commented on frontend/src/concepts/connectionTypes/ConnectionTypeForm.tsx:
```diff
@@ -176,6 +177,16 @@ const ConnectionTypeForm: React.FC<Props> = ({
         {connectionType && (
           <ConnectionTypeDetailsHelperText connectionType={connectionType} isPreview={isPreview} />
         )}
+        {connectionTypeName === 'oci-v1' && (
+          <Alert
+            variant="warning"
+            style={{ marginTop: 'var(--pf-t--global--spacer--lg)' }}
+            title="Deploying from an OCI connection"
+          >
+            Only models that are packaged as ModelCars can be deployed using OCI connections. Ensure
+            that that your model is in ModelCar format before deploying or the deployment will fail.
+          </Alert>
+        )}
```
Ah, so this is going to appear in every instance of the connectionTypeForm, even if it's not model serving specific. Which might not be ideal. It's tricky though since it's right in the middle of the form. you could pass the alert in as a prop. But maybe it's fine if it's always shown??

Also it would probably be better to change `connectionTypeName === 'oci-v1'` to `isModelServingCompatible(connectionType, ModelServingCompatibleTypes.OCI)`. Unless we only want to show it for the ootb connection type?

simrandhaliw commented:
@vconzola, just to confirm, we decided that we won’t display the modelcar's yellow warning alert in the 'Create Connection' modal, but instead, we will show it within the Connection section (for both 'Existing Connection' and 'Create Connection' options) when OCI connections are selected. So, the yellow warning in the first screenshot above should be removed, right?

emilys314 commented:
> Also, the info alert we display for 'Secret Details' is missing (see designs [here](https://www.figma.com/design/97akqulXiGIGtvxkdPLjMm/Data-connections-enhancement?node-id=5539-113679&t=PBc5AY7SrdqgCESh-4)). The microcopy for that might actually change a bit. I’ll send an update as soon as I hear back from Katie.

@simrandhaliw that part is here https://github.com/opendatahub-io/odh-dashboard/pull/3868

Griffin-Sullivan commented:
Ok thanks I'll fix the warning and look into the italics. I'm also going to add the warning for the model URI when it doesn't match the registry host here since it's basically microcopy.

vconzola commented:
@Griffin-Sullivan Just so we're all on the same page, the final microcopy for the warning to be shown within the Connection section (for both 'Existing Connection' and 'Create Connection' options) when OCI connections are selected is:
"**Model requirement for OCI connections**
OCI connections can be used to deploy only models packaged as ModelCars. Deploying models packaged in different formats will result in failure."

vconzola commented:
@Griffin-Sullivan The warning alert lgtm. Not sure about other microcopy changes. @simrandhaliw will have to weigh in on those.

Griffin-Sullivan commented:
> except for the 'View connection type details' link in the Create connection section. It should be styled similarly to the 'info icon + View details' link displayed in the Existing connection section.

@simrandhaliw I can do that too wasn't sure if we wanted to since it wasn't mentioned in the figma. I'll update now!

Griffin-Sullivan commented:
![Screenshot From 2025-03-13 11-21-46](https://github.com/user-attachments/assets/8310582a-0343-4926-ae4b-7b452c358b50)
![Screenshot From 2025-03-13 11-21-59](https://github.com/user-attachments/assets/8ebd29de-9341-4d52-bb47-d8dd81aa5c00)

@simrandhaliw here's the button for new connections

simrandhaliw commented:
Thank you! LGTM!

emilys314 commented on frontend/src/concepts/connectionTypes/ConnectionTypeForm.tsx:
```diff
@@ -128,6 +129,7 @@ const ConnectionTypeForm: React.FC<Props> = ({
   onChange,
   onValidate,
   connectionErrors,
+  Alert = undefined,
```
Very minor nitpick, but should this start with lowercase since it's a prop variable? I guess it's a component prop, but I'm not sure which priority. Thoughts?

Also imo you don't have to init an optional prop to undefined, the `?` will make it for you, it saves code clutter

Griffin-Sullivan replied:
Yeah I went back and forth but landed on capital. I don't really care though. And yeah I can do ?

emilys314 replied:
You don't have to change it then

emilys314 commented on frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ConnectionOciPathField.tsx:
```diff
@@ -8,19 +8,25 @@ import {
   setModelUri,
+  isNewConnection = false,
 }) => {
+  const [isOciHostInModelUri, setIsOciHostInModelUri] = React.useState(true);
```
Actually I wonder if we should make this a React.useMemo that checks on the fly. Because if you enter the oci URI, but then update the registry path, the 2 can get out of sync. Thoughts? (That does mean the onblur / onChange logic changes from what we talked about)

![image](https://github.com/user-attachments/assets/9231d42a-2c7a-4616-8914-e41925222beb)

Griffin-Sullivan replied:
ok I can switch back to useMemo but that means the warnings show up as you type / edit. Unless there's a way to fix it and use Memo?

emilys314 replied:
I don't think there is a clean / simple way to do that with just useMemo. you can resolve the convos as you see fit.

Griffin-Sullivan replied:
Decided to keep it as is but added the `onFocus` in case you tab / click on the uri path again it will take the warning away

emilys314 replied:
oo ok nice

emilys314 replied:
So I'm seeing an issue with the validation, when i focus and then blur, it seems to be acting differently 
![image](https://github.com/user-attachments/assets/4fb1e406-c14a-40a3-9e19-b2c4645f3270)

I think this might be due to the prefix that gets added `oci://`, 

Also I kinda thought you would validate on focus, not remove the error. Because if t's in the errors state and you go click inside of it, the error goes away, which is slightly confusing because a user might think it's valid again even if they didn't change the input

Griffin-Sullivan replied:
Updated this to remove the on focus

emilys314 replied:
There is still an issue when you repeatedly focus and blur the text input. I think it's from the oci:// prefix behind the scenes. you may want you validation to ignore this, or change it to `includes()`? 

![image](https://github.com/user-attachments/assets/b389388f-a84a-4f62-8a75-cd720af55494)

Griffin-Sullivan commented:
@simrandhaliw just realized when I added that view details button it also changed it for the Create Connection modal. Is that how it should be?
![image](https://github.com/user-attachments/assets/e2017685-101c-47b6-87ca-81a8798bd344)

simrandhaliw commented:
@Griffin-Sullivan Yes, that update LGTM! Any modal that previously had the link buttons below the dropdown for 'View connection details' or 'View connection type details' should now ideally be replaced with the info icon + 'View details' button on the same line as the dropdown.

emilys314 commented:
/lgtm
/approve
